### PR TITLE
Release 1.4

### PR DESCRIPTION
**Pull request by p516entropy**
Improved atg-build-plugin.

Added: module dependencies management like ATG dependencies.
Added: manifest file generation task based on gradle.build.
Added: applied dependencies jar fetcher task.
Added: supports ATG-Required-If-Present dependencies.
Added: replaced atg configuration prefix to atgRequired and atgRequiredIf.
Added: several small fixes

### CODE REVIEW

File: src/main/groovy/com/epam/atg/gradle/manifest/ManifestConfig.groovy [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    @Input Set<String> atgClassPath = []
+    @Input Set<String> atgRequired = []
+    @Input Set<String> atgRequiredIfPresent = []
+    @Input @Optional String atgConfigPath
+    @Input Set<String> projectClassPath = []
+    @Input boolean generateAtgClientClassPath = false
+    @Input boolean generateIndividualResources = false
+    @Input boolean skipGeneration = true
+    @Input boolean override = false
+    @Input String manifestFilePath = "META-INF/MANIFEST.MF.new"
```
Comment (prokofiev) [inline src/main/groovy/com/epam/atg/gradle/manifest/ManifestConfig.groovy]: what is it? why ".new"

**Review by prokofiev — CHANGES_REQUESTED** [review]

File: examples/TProjectManifestGeneration/ModuleA/src/com/tproject/nucleus/NucleusUtils.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+	public boolean checkNucleus() {
+		/*
+		atg.nucleus.Nucleus nucleus = atg.nucleus.Nucleus.getSystemNucleus();
+		return nucleus != null;
+		*/
+		return false;
+	}
+
+}
```
[the remark this answers is not in the document]
Comment (Aliaksei-Tatarynchyk) [reply examples/TProjectManifestGeneration/ModuleA/src/com/tproject/nucleus/NucleusUtils.java]: why the new line at the end of file is needed?

[the remark this answers is not in the document]
Comment (prokofiev) [reply examples/TProjectManifestGeneration/ModuleA/src/com/tproject/nucleus/NucleusUtils.java]: This is not mandatory but is considered a good style of writing code.

**Review by wenditer — APPROVED** [review]

File: examples/TProjectManifestGeneration/ModuleA/src/com/tproject/nucleus/NucleusUtils.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Aliaksei-Tatarynchyk) [reply examples/TProjectManifestGeneration/ModuleA/src/com/tproject/nucleus/NucleusUtils.java]: I've heard only about mandatory adding of new line to the end of manifest files, never heard the same in context of groovy or java code styles. could you please give me some references/links on it?

**Review by prokofiev — APPROVED** [review]

**Pull request merged**